apps/dashboard/services/graphql/index.ts

```diff
@@ -1,17 +1,36 @@
-import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
-import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";
 import { env } from "@/env";
+import {
+  ApolloClient,
+  HttpLink,
+  InMemoryCache,
+  ApolloLink,
+} from "@apollo/client";
+import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";
+
+const headerSettingLink = new ApolloLink((operation, forward) => {
+  const request = operation.getContext().request;
+  operation.setContext({
+    headers: {
+      "x-real-ip": request?.headers.get("x-real-ip") || "",
+      "x-forwarded-for": request?.headers.get("x-forwarded-for") || "",
```

Comment:
do we really want to default to `""` - isn't it better to leave it undefined? cc @nicolasburtey

Reply:
agree but this may have to do with type?

Reply:
yes